hs/src/Reach/Eval/Core.hs

```diff
@@ -2611,6 +2611,12 @@ warnInteractType = \case
 evalPrim :: SLPrimitive -> [SLSVal] -> App SLSVal
 evalPrim p sargs =
   case p of
+    SLPrim_castOrTrunc to -> do
+      let (trunc, sargs') =
+              case sargs of
+                [i, (_, SLV_Bool _ trunc_)] -> (trunc_, [i])
+                _ -> (False, sargs)
```

Comment:
This code can be cleaner. I think it should be like `SLPrim_Token_Burn` and then after do a `fromMaybe False mtrunc`